core/src/main/java/org/infinispan/cache/impl/CacheImpl.java

```diff
@@ -1178,6 +1178,10 @@ final void putAll(Map<? extends K, ? extends V> map, Metadata metadata, long exp
    private void putAllInternal(Map<? extends K, ? extends V> map, Metadata metadata, long explicitFlags, InvocationContext ctx) {
       InfinispanCollections.assertNotNullEntries(map, "map");
       Metadata merged = applyDefaultMetadata(metadata);
+      // Vanilla PutMapCommand returns previous values; add IGNORE_RETURN_VALUES as the API will drop the return value
+      // interceptors are free to clear this flag if appropriate (since interceptors are the only consumers of the
+      // return value)
```

Comment:
And that's why we set the flag by default, and interceptor request them by clearing the flag. I could introduce another flag `FORCE_RETURN_VALUE` but this would pollute the code without any gain.